openverse_catalog/dags/common/dag_factory.py

```diff
@@ -22,19 +32,64 @@
     "on_failure_callback": slack.on_failure_callback,
 }
 DATE_RANGE_ARG_TEMPLATE = "{{ macros.ds_add(ds, -%s) }}"
+XCOM_PULL_TEMPLATE = "{{ ti.xcom_pull(task_ids='%s', key='%s') }}"
+
+
+def _push_output_paths_wrapper(
+    func: Callable,
+    media_types: List[str],
+    ti: TaskInstance,
+    args: Sequence = None,
+):
+    """
+    Run the provided callable after pushing the calculated output directories
+    for each media store to XComs. Output locations are pushed under keys with
+    the format `<media-type>_tsv`. This is a temporary workaround due to the nature
+    of the current provider scripts. Once
+    https://github.com/WordPress/openverse-catalog/issues/229 is addressed and the
+    provider scripts are refactored into classes, this wrapper can either be updated
+    or the XCom pushing can be moved into the provider initialization.
+    """
+    args = args or []
+    logger.info("Pushing available store paths to XComs")
+    module = inspect.getmodule(func)
+    stores = {}
+
+    # Stores exist at the module level, so in order to retrieve the output values we
+    # must first pull the stores from the module.
+    for media_type in media_types:
+        if not (store := getattr(module, f"{media_type}_store", None)):
+            continue
+        stores[media_type] = store
+
+    if len(stores) != len(media_types):
+        raise ValueError(
+            f"Expected a store of the following types in {module.__name__} "
+            f"but none were found: {media_types}"
+        )
+
+    for media_type, store in stores.items():
+        logger.info(f"{media_type.capitalize()} store location: {store.output_path}")
+        ti.xcom_push(key=f"{media_type}_tsv", value=store.output_path)
+
+    logger.info("Running provider function")
+    # Not passing kwargs here because Airflow throws a bunch of stuff in there that none
+    # of our provider scripts are expecting.
+    return func(*args)
 
 
 def create_provider_api_workflow(
-    dag_id,
-    main_function,
-    default_args=None,
-    start_date=datetime(1970, 1, 1),
-    max_active_tasks=1,
-    schedule_string="@daily",
-    dated=True,
-    day_shift=0,
-    dagrun_timeout=timedelta(hours=12),
-    doc_md="",
+    dag_id: str,
+    main_function: Callable,
+    default_args: Optional[Dict] = None,
+    start_date: datetime = datetime(1970, 1, 1),
+    max_active_tasks: int = 1,
+    schedule_string: str = "@daily",
+    dated: bool = True,
+    day_shift: int = 0,
+    dagrun_timeout: timedelta = timedelta(hours=12),
+    doc_md: str = "",
+    media_types: List[str] = ("image",),
```

Comment:
Does `List` type also include tuples? I understand that the default value cannot be a list because it's mutable. Does the type declaration match the default value?

Reply:
Ugh, even after [all my best intentions](https://github.com/WordPress/openverse-catalog/pull/357#discussion_r809458638) I still get this type wrong! I'll fix this to make it more generic.